src/App.js

```diff
@@ -14,11 +15,90 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    initialGoods: goodsFromServer,
+    isListVisible: false,
+    isReversed: false,
+    sortBy: '',
+  }
+
+  showList = () => {
+    this.setState(state => ({
+      isListVisible: !state.isListVisible,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortByAlphabet = () => {
+    this.setState({ sortBy: 'alphabet' });
+  }
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  }
+
+  reset = () => {
+    this.setState(state => ({
+      sortBy: '',
+      isReversed: false,
+    }));
+  }
+
+  render() {
+    const { initialGoods, isListVisible, isReversed, sortBy } = this.state;
+
+    const goods = [...initialGoods].sort((item1, item2) => {
+      switch (sortBy) {
+        case 'alphabet':
+          return item1.localeCompare(item2);
+
+        case 'length':
+          return item1.length - item2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      goods.reverse();
+    }
+
+    return (
+      <div className="App">
+        <div className="choose-page">
+          <h1 className="page-title">Goods</h1>
+          <button
+            className="btn start-btn"
+            type="button"
+            onClick={this.showList}
+          >
+            {isListVisible
+              ? 'Hide'
+              : 'Start'
+            }
+          </button>
+          <div className="goods-block">
+            {isListVisible && (
+              <GoodsList
+                goods={goods}
+                reverse={this.reverse}
+                sortByAlphabet={this.sortByAlphabet}
+                sortByLength={this.sortByLength}
+                reset={this.reset}
```

Comment:
if you already sort an array why do you need to pass all those props, I think goods should enough.